src/main/java/ArraySortedBag.java

```diff
@@ -0,0 +1,193 @@
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class ArraySortedBag<T extends Comparable<? super T>> implements SortedBag<T> {
+
+    private static final int DEFAULT_CAPACITY = 100;
+    private static final int NOT_FOUND = -1;
+    protected T[] bag;
+    protected int rear;
+
+    public ArraySortedBag() {
+        this(DEFAULT_CAPACITY);
+    }
+
+    @SuppressWarnings("unchecked")
+    public ArraySortedBag(int initialCapacity) {
+        bag = (T[]) new Object[initialCapacity];
+        rear = 0;
+    }
+
+    /**
+     * Doubles the size of the bag array and copy the
+     * contents over.
+     */
+    @SuppressWarnings("unchecked")
+    private void expandCapacity() {
+        T[] newBag = (T[]) new Object[bag.length * 2];
+        for (int i = 0; i < bag.length; ++i) {
+            newBag[i] = bag[i];
+        }
+        bag = newBag;
+    }
+
+    /**
+     * Shifts elements in an array down (towards index 0) towards
+     * the starting index specified. This method assumes that
+     * the calling function manages the rear field.
+     *
+     * @param start Index of where shift starts/stops and element
+     *              overwritten.
+     */
+    private void shiftLeft(int start) {
+        for (int i = start; i < bag.length - 1; ++i) {
+            bag[i] = bag[i + 1];
+        }
+        bag[bag.length - 1] = null;
+    }
+
+    /**
+     * Shifts elements in an array up (towards index rear) away
+     * from the starting index specified. This method assumes that
+     * the calling function manages the rear field and any
+     * need of expandCapacity().
+     *
+     * @param start Index of where shifting starts/stops and where
+     *              space is opened.
+     */
+    private void shiftRight(int start) {
+        for (int i = rear; i > start; --i) {
+            bag[i] = bag[i - 1];
+        }
+        bag[start] = null;
+    }
+
+    /**
+     * Find the index of a specified target element. If the element
+     * does not exist, return NOT_FOUND
+     *
+     * @param target Element to be found
+     * @return Index of the element if found, NOT_FOUND otherwise
+     */
+    private int indexOf(T target) {
+        for (int i = 0; i < bag.length; ++i) {
+            if (bag[i].equals(target)) {
+                return i;
+            }
+        }
+        return NOT_FOUND;
+    }
```

Comment:
Could be a pretty pogged intro to searching/sorting. Bad implementation now, revisit later.

Reply:
Gets hit when we talk about binary search trees.